Choosing between javascript and http communication.

tommythorsen opened the pull request:
This changes the spec proposal so that a payment app can decide whether
it wants to communicate with the user agent by means of HTTP or by means
of javascript functions.

The choice is implemented as an additional manifest field called
communication_method.

See issue #130

Commit:
Choosing between javascript and http communication.

This changes the spec proposal so that a payment app can decide whether
it wants to communicate with the user agent by means of HTTP or by means
of javascript functions.

The choice is implemented as an additional manifest field called
communication_method.

dlongley commented:
Maybe the app should declare its supported protocols instead and let the mediator decide which is best for its platform/context.

ianbjacobs commented:
+1 to merging the PR (mostly to review it; not ready to endorse the proposal though yet).

Thanks!

Ian

Commit:
Change the communication_methods member into a list.

This lets the payment app declare which communication methods it
supports so that the user agent can choose which method to use.

tommythorsen commented:
@dlongley 

> Maybe the app should declare its supported protocols instead and let the mediator decide which is best for its platform/context.

This sounds like a good idea. I pushed another commit that does this.

adrianhopebailie commented:
Based on discussion in the TF we want to pick one way to do this and demonstrate that it is easy to acheive the other through a simple bridge (i.e. if we use HTTP then a JavaScript service worker can provide an offline JavaScript implementation and if we use JS then it can send messages to a remote service)
